src/agent/solana/oracle.rs

```diff
@@ -593,12 +661,13 @@ impl Poller {
             // as todo gets replaced with next_todo.
             for (price_key, price_account) in todo.iter().zip(price_accounts) {
                 if let Some(price_acc) = price_account {
-                    let price = load_price_account(&price_acc.data)
+                    let price = PriceEntry::load_from_account(&price_acc.data)
                         .context(format!("Could not parse price account at {}", price_key))?;
 
+                    let next_price = price.next;
```

Comment:
Do we need this extra variable?

Reply:
Yeah unfortunately the price gets moved above, rather than clone it this just copies out the only useful data we need.